depdive/src/update.rs

```diff
@@ -96,6 +97,22 @@ pub enum VersionConflict {
     },
 }
 
+#[derive(Debug, Clone)]
+pub struct VersionUnsafeDiffStats {
+    pub files_changed_containing_unsafe_code: u64,
+    pub files_requiring_review: Vec<String>, // Files potentially containing unsafe change
+    pub files_unsafe_change_stats: Vec<FileUnsafeChangeStats>,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum UnsafeCodeChangeStatus {
+    UnsafeCounterModified, // when we have a delta in unsafe counter
+    NoUnsafeCode,          // changed file(s) contained no unsafe code before and after change
+    UnsafeCodeRemoved,     // there was unsafe code before the change that got removed
+    RequiresManualReview,  // changed files contain unsafe code,
```

Comment:
Can we come up with a new name that doesn't imply a specific review policy?

Also is `UnsafeCodeRemoved` mean that all unsafe was removed or just some? Even if unsafe code is removed if it is not all the unsafe code in a module, we can't know the result is more safe without manually reviewing the code for invariant violations.

Reply:
`UnsafeCodeRemoved` is `AllUnsafeCodeRemoved` - will change that. 
`RequiresManualReview` can be renamed as `Uncertain`.

Reply:
Uncertain is better in that now it doesn't imply some policy, but is there not a more descriptive name that tells what the objective metric here is?

Reply:
Perhaps @bmwill has some ideas?

Reply:
@metajack, The `uncertain` part is basically a todo here. something that needs to be revisited and implemented. I am not seeing it as a metric. For now, `UnsafeCounterModified` and `Uncertain` both can be seen together as one metric that tells a file containing unsafe code has been changed in the diff.

Reply:
I'm having difficulty understanding the difference between Uncertain and UnsafeCounterModified, both seem to indicate that unsafe could have been changed?

Reply:
`UnsafeCounterModified` : We found a delta in the unsafe counter before and after a change of a file which indicates we are certain that there was changes made to the unsafe code.

`Uncertain`  is basically when we did not find a delta, the unsafe counter remains same before and after the change but the files contains some unsafe code. In this case, we can basically say nothing more than this. We don't know if any unsafe code has been modified or not.

Reply:
See this comment: https://github.com/diem/whackadep/pull/98#discussion_r671372590 maybe we can do something more?

Reply:
> both seem to indicate that unsafe could have been changed?
yes